spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java

```diff
@@ -190,9 +190,10 @@ protected Object doReceive() {
 
 					if (this.toBeReceived.size() > 0) { // don't re-fetch already filtered files
 						file = poll();
+						continue;
 					}
 					else {
-						file = null;
+						break;
```

Comment:
Grate!

I think we don't need an `if (file != null)` any more after this.

Reply:
Sure thing will do tomorrow

Reply:
@artembilan test added